### PR TITLE
Reflection

### LINKED ISSUE #65: Reflection on the product and process from a software engineering perspective
Part of the report

### PR DESCRIPTION
**Pull request by RolfStarre**
Reflection on the product and process from a software engineering perspective. Fixes #65

### CODE REVIEW

File: Final Report/Report.tex [hunk 1]
```diff
@@ -97,6 +97,24 @@ \subsubsection*{Will not haves}
...
+In the Runtime repository the emotions had to be handled during runtime. This was done by connecting the adding and dropping of goals to the Gamygdala engine and by making sure that the emotion updates from the Gamydala engine were handled and added to a base similar to how percepts are updated every cycle. We implemented this by adding calls in the existing code to functions that we made to handle the emotions. For the structure of our functions we looked at how the existing code handled similar situations. After some initial time to get to know the existing code base we were able to implement the functionality we wanted in this repository relatively easy. However, when we tried to add to add an EmotionBase rather than use the PerceptBase to process the emotion we learned that this was a lot harder to do. \par
```
Comment (mspanoghe) [inline Final Report/Report.tex]: Ik denk dat het moet zijn:
"After some initial time to get to know the existing code base we were able to implement the functionality we wanted in this repository relatively easily(easy stond er)."
Ben niet zeker hoor.

File: Final Report/Report.tex [hunk 2]
```diff
@@ -97,6 +97,24 @@ \subsubsection*{Will not haves}
...
+
+\subsection{Process}
+We made use of an incremental and iterative software development process. Every week we made a scrum plan and at the end of the week we reflected on this plan and made a new one. Initially the tasks we made were not very fitting for scrum, they were somewhat too generic and did not lead to something we could demo. We were able to use the feedback given by the TA and the problems we encountered in our weekly scrums to improve our scrum process over the course of the project. \par 
+We used git on the social platform Github in combination with TravisCI and Maven for our code versioning and continuous integration. The code bases we used were already on github and after cloning them to our own repositories we later had some troubles with the POM-files which were still referring to the original repositories. This problem came back several time, and we should probably have fixed this initially so any new issues would be smaller and therefor easier to solve. \par
```
Comment (mspanoghe) [inline Final Report/Report.tex]: came back several timeS*

**Comment by mspanoghe** [comment]
Top!

File: Final Report/Report.tex (hunk 2 above)
Comment (jwooning) [inline Final Report/Report.tex]: "The code bases we used were already on github and after cloning them to our own repositories"

Dit moet forking zijn ipv cloning?

File: Final Report/Report.tex [hunk 3]
```diff
@@ -97,6 +97,24 @@ \subsubsection*{Will not haves}
...
+Another problem we encountered here was that the original code already failed when doing integration tests, this had to do with the tests trying to open windows which was not possible and to solve this we had to ignore a few tests. Regarding testing, the original code lacked good test coverage and integration testing. Due to the difficulty of integration testing the whole code base including our new code we decided to focus our testing efforts on making unit tests involving the code we added to the existing code. However, the lack of integration tests available often make it hard to find where errors originated.  \par 
+This was also the first project where we made use of a pull-based development model. Initially we made some mistakes by pulling changes in the master onto our own base instead of rebasing. We also learned to squash a lot of small changes into fewer more substantial commits to make it easier to track the changes that were made. 
+\clearpage
+
```
Comment (jwooning) [inline Final Report/Report.tex:118]: Misschien ook erbij zetten dat de originele codebase niet bruikbaar was voor checkstyle of goede javadocs

**Comment by jwooning** [comment]
Verder goed

**Pull request merged**